contracts/abstracts/registrarabstract.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity ^0.4.0;
+
+import "./trialrulesabstract.sol";
+import "./token.sol";
+import "./owned.sol";
+
+
+contract RegistrarAbstract is Owned {
+
+
+    /// @notice register - register a player to the game
+    ///
+    /// The function will throw if the player is already register or there is not
+    /// enough deposit in the contract to ensure the player could be compensated for the
+    /// case of a valid case.
+    /// @param _player  - the player address
+    /// @return bool registered - true for success registration.
+    function register(address _player) onlyOwner public returns (bool);
+
+    /// @notice deposit - deposit to the contract
+    ///
+    /// @param epochs  - deposit epochs
+    /// @return bool registered - true for success deposit.
+    function deposit(uint epochs) payable returns (bool);
+
+    /// @notice collectDeposit - collect back the caller deposit.
+    ///
+    ///The function check that there is no open case for the specific caller.
+    ///The function check that there is enough deposit left in thec contract for the case of a valid case compensation.
+    /// @return bool  true for success otherwise false.
+    function collectDeposit() external returns (bool);
+
+    /// @notice isRegistered - Check if a player is registered to the game
+    ///
+    /// @param player  - player address
+    /// @return bool  true for success otherwise false.
+    function isRegistered(address player) returns (bool);
+
+    /// @notice compensate - compensate the beneficiary with the reward amount
+    ///
+    /// @param _beneficiary  - beneficiary address
+    /// @param reward        - reward amount
+    /// The function will throw if it is not called by the swear contract.
+    /// @return bool  true for success otherwise false.
+    function compensate(address _beneficiary,uint reward) returns(bool compensated);
```

Comment:
onlyOwner will not work here because msg.sender will be the swear contract address and not the owner of the registrar contract .
The service (which is the owner) should not call this function directly.